platform/linuxbsd/display_server_x11.cpp

```diff
@@ -1173,7 +1173,7 @@ void DisplayServerX11::window_set_position(const Point2i &p_position, WindowID p
 			unsigned char *data = nullptr;
 			if (XGetWindowProperty(x11_display, wd.x11_window, prop, 0, 4, False, AnyPropertyType, &type, &format, &len, &remaining, &data) == Success) {
 				if (format == 32 && len == 4 && data) {
-					long *extents = (long *)data;
+					int64_t *extents = (int64_t *)data;
 					x = extents[0];
```

Comment:
This will go badly wrong if your assumption is wrong (say some 32 bit version, or quirky OS).

If an API is returning a long by value, you can usually convert it to 64 bit safely, as it will either be converting 32 to 64 bit, or be a no-op. But with pointers, you can't do this safely, because it relies on the data being that size.

This can potentially lead to extremely hard to find bugs. Isn't this why c style casting is discouraged and the use of `static_cast` is encouraged? `static_cast` will flag a compile error in these cases. I tend to use c style casting when I'm sure, but in cases like this `static_cast` would be *much* better.

Reply:
The `format == 32 && len == 4` is also highly suspicious.

Reply:
The reference for [XGetWindowProperty][1] says:

> If the returned format is **8**, the returned data is represented as a **char** array. If the returned format is **16**, the returned data is represented as a **short** array and should be cast to that type to obtain the elements. If the returned format is **32**, the returned data is represented as a **long** array and should be cast to that type to obtain the elements.

I interpret that as "use long regardless of how large it really is". I'm not sure if that assumption is correct though.

As for the sketchy `format == 32 && len == 4`, [_NET_FRAME_EXTENTS][2] says:

>  _NET_FRAME_EXTENTS, left, right, top, bottom, CARDINAL[4]/32

I interpret that as "always 4 values with format 32", so I suppose the check isn't needed.

[1]: https://www.x.org/releases/X11R7.6/doc/man/man3/XChangeProperty.3.xhtml
[2]: https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#idm44946547109312